R/calculate_stay.R

```diff
@@ -0,0 +1,32 @@
+#' Calculate total length of stay
+#'
+#' @description Calculate the total length of stay between start_date and end_date.
+#' If the end_date is missing then use the dummy discharge date.
+#'
+#' @param year The financial year in '1920' format
+#' @param start_date The admission/start date variable. e.g. record_keydate1
+#' @param end_date The discharge/end date variable. e.g record_keydate2
+#' @param sc_qtr The latest submitted quarter. e.g. sc_latest_submission
+#'
+#' @return a [tibble][tibble::tibble-package] with additional variable `stay`.
+#' If there is no end date use dummy discharge to calculate the total length of stay.
+#' @export
+#'
+#' @family date functions
+calculate_stay <- function(year, start_date, end_date, sc_qtr = NULL) {
+  lubridate::time_length(lubridate::interval(start_date, end_date), unit = "days")
+
+  if (is.na(end_date) & is.null(sc_qtr)) {
```

Comment:
There may be a warning message indicating "the condition has length > 1", which is because is.null() returns a scalar result rather than a vector result.

Reply:
Maybe something like:
```
if (is.missing(sc_qtr)) {
  ...
}
```